lua/scratch/api.lua

```diff
@@ -8,6 +8,7 @@ local MANUAL_INPUT_OPTION = "MANUAL_INPUT"
 ---@class Scratch.ActionOpts
 ---@field window_cmd? Scratch.WindowCmd
 ---@field content? string[] content will be put into the scratch file
+---@field hooks? {[string]: Scratch.Hooks, [Scratch.Trigger]:Scratch.Hook}
```

Comment:
`[string]: Scratch.Hooks` it is because after choosing, you need to call correctly.. I think another approach is `{[Scratch.Trigger]:table<string, Scratch.Hook>| Scratch.Hook}`

Each of the ideas is disgusting in its own way